src/components/backstop/BackstopDepositAnvil.tsx

```diff
@@ -87,15 +79,17 @@ export const BackstopDepositAnvil: React.FC<PoolComponentProps> = ({ poolId }) =
             <InputBar
               symbol={'BLND-USDC LP'}
               value={toDeposit}
-              onValueChange={handleDepositAmountChange}
+              onValueChange={setToDeposit}
               onSetMax={handleDepositMax}
               palette={theme.palette.backstop}
               sx={{ width: '100%' }}
+              isMaxDisabled={!userBalance}
             />
             <OpaqueButton
               onClick={handleSubmitTransaction}
               palette={theme.palette.backstop}
               sx={{ minWidth: '108px', marginLeft: '12px', padding: '6px' }}
+              disabled={!userBalance || !toDeposit || Number(toDeposit) > userBalance}
```

Comment:
not sure I am a massive fan of the logic being placed here for all anvils

I think it will end up getting moved up once we attribute tooltips to the disabled states, so I'm not gonna block this PR for the code style change.

Reply:
moved it